### PR TITLE
Test password utils

### PR DESCRIPTION
**Pull request by williamcroberts**
This PR drops the -X aka hex password option and allows the specification of formatting in the password string itself. For commands that take multiple passwords, this allows to mix and match hex and string passwords. Before it was just one or the other.

This also gets rid of the crufty conversion/copy routines in password utils and adds unit tests.

This brings issue #321 to a close.

Fixes: #321

### CODE REVIEW

File: test/unit/test_tpm2_password_util.c [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    (void)state;
+
+    TPM2B_AUTH dest;
+    /* 65 hex chars generated via: echo \"`xxd -p -c256 -l65 /dev/urandom`\"\; */
+    char *overlength =
+        "ae6f6fa01589aa7b227bb6a34c7a8e0c273adbcf14195ce12391a5cc12a5c271f62088"
+        "dbfcf1914fdf120da183ec3ad6cc78a2ffd91db40a560169961e3a6d26bf";
+    bool res = tpm2_password_util_from_optarg(overlength, &dest);
+    assert_false(res);
+}
```
Comment (williamcroberts) [inline test/unit/test_tpm2_password_util.c]: Needs a test for the empty string

File: test/unit/test_tpm2_password_util.c [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    TPM2B_AUTH dest = {
+        .t = { .size = 42 }
+    };
+
+    bool res = tpm2_password_util_from_optarg("str:", &dest);
+    assert_true(res);
+    assert_int_equal(dest.t.size, 0);
+}
+
+static void test_tpm2_password_util_from_optarg_empty_str_str_prefix(void **state) {
```
Comment (williamcroberts) [inline test/unit/test_tpm2_password_util.c]: Names are flipped around for tests, this one should be testing "str:"
